### ISSUE TITLE
Upgrading to 2.0 - Resolver must have required property 'functions'

### ISSUE THREAD

**Issue by sparksterz** [sid88in/serverless-appsync-plugin#568]
Hi, I've recently begun looking into updating my Appsync APIs from our node v14.x runtime to v18.x and migrating to the latest serverless and appsync plugins while doing so. I've been trying to keep things 1:1 with what's already deployed and have read through the migration document a few times. 

I think the error I'm getting stuck on seems to be due to the undocumented required property of 'functions' per the [resolver document](https://github.com/sid88in/serverless-appsync-plugin/blob/master/doc/resolvers.md) . My current AWS Appsync is actually deployed with no registered GraphQL functions, but instead just registered AWS_LAMBDA data sources for each GraphQL query. I couldn't find a supporting way to configure my API in this manner as omitting the functions section from the resolver results in the following error on `sls package`:
```
Error:
Invalid AppSync Configuration:
     at appSync/resolvers/Query.MY_RESOLVER: must have required property 'functions'
```

This is the CloudFormation error I got when trying to specify a "functions" section:
```The code contains one or more errors. (Service: AmazonDeepdish; Status Code: 400; Error Code: BadRequestException)```

Specifically when trying to create the graphql functions:
```The following resource(s) failed to create: [GraphQlFunctionConfigurationQueryMY_FIRST_RESOLVER0, ... ]. The following resource(s) failed to update: [GraphQlSchema].```

My V1 config looked like this:
```
functions:
  MY_FIRST_RESOLVER:
    handler: src/handlers/MY_FIRST_RESOLVER.MY_FIRST_RESOLVER
    timeout: 10
...
custom:
  appSync:
    ...
      mappingTemplates:
        - dataSource: MY_FIRST_RESOLVER
          type: Query
          field: MY_FIRST_RESOLVER
          request: Query.MY_FIRST_RESOLVER.request.vtl
          response: Query.MY_FIRST_RESOLVER.response.vtl
      dataSources:
        - type: AWS_LAMBDA
          name: MY_FIRST_RESOLVER
          description: Get Info for my resolver
          config:
            functionName: MY_FIRST_RESOLVER
```

I had then tried to define each resolver without a `functions` object only to get the error specified above: `at appSync/resolvers/Query.MY_RESOLVER: must have required property 'functions'`
So instead I transformed my V2 config while attempting to add a 'functions' section that basically repeats exactly what my top level `serverless.yml` section already outlines. When I deployed it that way is when I got my CloudFormation error. My config looked like this:
```
functions:
  MY_FIRST_RESOLVER:
    handler: src/handlers/MY_FIRST_RESOLVER.MY_FIRST_RESOLVER
    timeout: 10
...
appSync:
  resolvers:
      Query.MY_FIRST_RESOLVER:
        dataSource: MY_FIRST_RESOLVER
        request: ./mapping-templates/Query.MY_FIRST_RESOLVER.request.vtl
        response: ./mapping-templates/Query.MY_FIRST_RESOLVER.response.vtl
        functions:
          - dataSource: 'MY_FIRST_RESOLVER'
            code: 'src/handlers/MY_FIRST_RESOLVER.ts'
  dataSources:
      MY_FIRST_RESOLVER:
        type: AWS_LAMBDA
        description: Get Info for my resolver
        config:
          functionName: MY_FIRST_RESOLVER
```

My initial thinking via what documentation was that the [serverless documented 'functions'](https://www.serverless.com/framework/docs/providers/aws/guide/functions) is what the V2 dataSource 'functionName' refers to, and that seems confirmed by the documentation in this repo [here](https://github.com/sid88in/serverless-appsync-plugin/blob/master/doc/dataSources.md#config-1), but I can't tell if it's just out of date, or if I'm doing something wrong. 

In short would the combination of my Query resolver, specifying a specific dataSource, having the dataSource point to my top level serverless function via name, be something that's supported? It feels to me like it satisfies specifying the "function" but I can't see a way to satisfy it.

**Comment by bboure** [comment]
Hi,

The `functions` property refers to PIPELINE resolvers, which is something you don't seem to do.
The new default kind of resolvers is now `PIPELINE` [as explained in the migration guide](https://github.com/sid88in/serverless-appsync-plugin/blob/master/doc/upgrading-from-v1.md#defaults-to-pipeline-and-javascript-resolvers).

If you want to keep using `UNIT` resolvers, you will need to add `kind: UNIT`.
I think it might be worth adding this in the guide. 👍 

```yml
functions:
  MY_FIRST_RESOLVER:
    handler: src/handlers/MY_FIRST_RESOLVER.MY_FIRST_RESOLVER
    timeout: 10
...
appSync:
  resolvers:
      Query.MY_FIRST_RESOLVER:
        kind: UNIT # <-- add this
        dataSource: MY_FIRST_RESOLVER
        request: ./mapping-templates/Query.MY_FIRST_RESOLVER.request.vtl
        response: ./mapping-templates/Query.MY_FIRST_RESOLVER.response.vtl

  dataSources:
      MY_FIRST_RESOLVER:
        type: AWS_LAMBDA
        description: Get Info for my resolver
        config:
          functionName: MY_FIRST_RESOLVER
```

But you can keep your lambda function at the root, like you have it now and reference it by name, like you did.

**Comment by andrew-sol** [comment]
@bboure  Please update the "Quick Start" section of this doc https://github.com/sid88in/serverless-appsync-plugin/blob/master/doc/resolvers.md#quick-start since it provides an invalid config.

**Comment by andrew-sol** [comment]
@bboure I'm new to AWS Serverless and just trying to figure it out.

The following config:
```yml
appSync:
  resolvers:
    Query.user:
      dataSource: myDataSource

    getPosts:
      type: Query
      field: getPosts
      dataSource: myDataSource
```
is invalid because the resolvers kind is `PIPELINE` (current default) so the `functions` property is required for both of them.
